src/connector/src/source/kafka/source/reader.rs

```diff
@@ -118,9 +123,14 @@ impl SplitReader for KafkaSplitReader {
 impl KafkaSplitReader {
     #[try_stream(boxed, ok = Vec<SourceMessage>, error = anyhow::Error)]
     pub async fn into_stream(self) {
-        if let Some(stop_offset) = self.stop_offset && stop_offset == 0{
-            yield Vec::new();
-            return Ok(());
+        if let Some(stop_offset) = self.stop_offset {
+            if let Some(start_offset) = self.start_offset && start_offset == stop_offset {
+                yield Vec::new();
+                return Ok(());
+            } else if stop_offset == 0 {
+                yield Vec::new();
+                return Ok(());
+            }
```

Comment:
it returns nothing. Consider refusing to create a reader in such case.

Reply:
If we refuse to create a reader and return a error, there are two option:
1. We directly report error if the range return nothing. This behavior seems weird because we will not report error if a table is empty. 
2. We compare the error and transfer it from bottom to top. Because `SplitReaderImpl::create` return as a AnyError, we compare the error by comparing string, I'm not sure whether is a good way.

Reply:
> We directly report error if the range return nothing.

I vote for this solution. Table is an internal concept, which is easy to handle but querying from kafka has more overhead than from table. If we know the result is empty deterministically, why do we waste resources on it?
@liurenjie1024 @fuyufjh what do you think?

Reply:
I find a problem: We will try_join_all!() the reader so if return a error directly, that will cause other reader create fail. https://github.com/risingwavelabs/risingwave/blob/6cacd3bcd5887c3b8ef4806ec289e789e6f4c95b/src/source/src/connector_source.rs#L271

So seems option 1 is also not easy to implement.🥵

I draft a implmentation for option 2, if we encouter a empty reader, InnerConnectorSourceReader will process the error and it will store a None in the reader field.

Reply:
We will create batch source executor each split which means that we can't create source executor if all split is empty and they both be filtered. So for now we directly return error in this case. 
This behavior maybe cause the problem: If we left join a table and a empty source, e.g.`select * from table left join empty_source` ,directly return error will cause error return. But this scenario is rare I think...

If we want the user to see the empty result instead of a error, maybe we can:
- create a empty batch vlaue executor

Reply:
I don't think we should return error here and it feels like abusing of error reporting to me since empty result is common in query. I think just return empty vec is good enough.